apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -817,20 +817,36 @@ internal_block_pending(_Config) ->
                 <<"BlockWithTxsHashes">>,
                 block_to_endpoint_map(PendingBlock)),
     ct:log("Expected pending block ~p", [ExpectedPendingTx]),
-    {ok, 200, PendingTxDefault} = get_internal_block_preset("pending",
-                                                             default),
-    {ok, 200, PendingTxHashes} = get_internal_block_preset("pending",
-                                                             false),
-    true = equal_block_maps(ExpectedPendingTx, PendingTxDefault), 
-    true = equal_block_maps(ExpectedPendingTx, PendingTxHashes), 
+    GetPending =
+        fun TryToGet(_Opt, Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Opt, Attempts) ->
+                case get_internal_block_preset("pending", Opt) of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} -> TryToGet(Opt, Attempts -1)
+                end
+        end,
+    {ok, PendingTxDefault} = GetPending(default, 10),
+    {ok, PendingTxHashes} = GetPending(false, 10),
+    ValidateKeys =
+        fun(Map1, Map2, Key) ->
+            true = maps:get(Key, Map1, not_found1) =:=
+                   maps:get(Key, Map2, not_found2)
+        end,
+    ValidateKeys(ExpectedPendingTx, PendingTxDefault, <<"prev_hash">>),
```

Comment:
I understand this may have a different prev_hash than ExpectedPendingTx.

Reply:
Maybe a text description of the aim of the test would help simplifying it?

Reply:
This should not have a different prev_hash (since it is not expected to mine a block durring that part of the test)

Reply:
The idea behind the test is to veirfy that we have a _pending like_ block being returned by the API. Since the block candidate changes really rapidly, the check if it is actually the block being mined is left to the common tests

Reply:
> This should not have a different prev_hash (since it is not expected to mine a block durring that part of the test)

Maybe writing this as a comment here would help reader.